### PR TITLE
use `EMBER_ENV` to configure build enviroment for `ember build`

### PR DESCRIPTION
**Pull request by hone**
This allows people to set non production environments, but still
defaults to production. See #3 for discussion.

You should be able to test it by setting your buildpack to:

```
$ heroku buildpacks:set https://codon-buildpacks.s3.amazonaws.com/buildpacks/hone/emberjs.tgz
```

### CODE REVIEW

File: buildpack/mrblib/buildpack/commands/compile.rb [hunk 1]
```diff
@@ -35,9 +37,9 @@ def run
 
           tuple =
             if dependencies["ember-cli-deploy"]
-              EmberBuildTuple.new(true, "ember deploy production", StaticConfig::DEFAULT_EMBER_CLI_DEPLOY_DIR)
+              EmberBuildTuple.new(true, "ember deploy #{@env["EMBER_ENV"]}", StaticConfig::DEFAULT_EMBER_CLI_DEPLOY_DIR)
```
Comment (fivetanley) [inline buildpack/mrblib/buildpack/commands/compile.rb]: Does this need to be shell escaped?

Comment (hone) [inline buildpack/mrblib/buildpack/commands/compile.rb]: yeah, good call. I imagine it'd be rare but better to be safe.

File: buildpack/mrblib/buildpack/commands/compile.rb [hunk 2]
```diff
@@ -35,9 +37,9 @@ def run
 
           tuple =
             if dependencies["ember-cli-deploy"]
-              EmberBuildTuple.new(true, "ember deploy production", StaticConfig::DEFAULT_EMBER_CLI_DEPLOY_DIR)
+              EmberBuildTuple.new(true, "ember deploy #{Shellwords.escape(@env["EMBER_ENV"])}", StaticConfig::DEFAULT_EMBER_CLI_DEPLOY_DIR)
             else
-              EmberBuildTuple.new(false, "ember build --environment production", StaticConfig::DEFAULT_EMBER_CLI_DIR)
+              EmberBuildTuple.new(false, "ember build", StaticConfig::DEFAULT_EMBER_CLI_DIR)
```
Comment (acorncom) [inline buildpack/mrblib/buildpack/commands/compile.rb:42]: Do we want to allow specifying an environment here as well?

Comment (hone) [inline buildpack/mrblib/buildpack/commands/compile.rb:42]: This should pick up environment based off of the env var `EMBER_ENV` on line 29, no?

Comment (acorncom) [inline buildpack/mrblib/buildpack/commands/compile.rb:42]: Confirmed, sorry about that

File: buildpack/mrblib/buildpack/commands/compile.rb [hunk 3]
```diff
@@ -35,9 +37,9 @@ def run
 
           tuple =
             if dependencies["ember-cli-deploy"]
-              EmberBuildTuple.new(true, "ember deploy production", StaticConfig::DEFAULT_EMBER_CLI_DEPLOY_DIR)
+              EmberBuildTuple.new(true, "ember deploy #{Shellwords.escape(@env["EMBER_ENV"])}", StaticConfig::DEFAULT_EMBER_CLI_DEPLOY_DIR)
```
Comment (akatov) [inline buildpack/mrblib/buildpack/commands/compile.rb:40]: this is not really EMBER_ENV - this is the *target* of `ember deploy`. These are orthogonal concepts and should be configured using separate environment variables.

**Comment by lancedikson** [comment]
Any updates here? Looks good, let's merge? @Dhaulagiri @hone

**Comment by hone** [comment]
@akatov how about `EMBER_ENV` and `EMBER_CLI_DEPLOY_TARGET`? It seems by default many people would prefer that they're the same. If `EMBER_CLI_DEPLOY_TARGET` is not set, it would equal `EMBER_ENV` or is that too confusing?

**Comment by akatov** [comment]
@hone I'm OK with `EMBER_CLI_DEPLOY_TARGET` defaulting to `EMBER_ENV`'s value, as long as you can override it. :-)

**Comment by dbinetti** [comment]
anything blocking this?

**Comment by Exelord** [comment]
Whats the status? :) This feature would be really welcome

**Comment by danilovaz** [comment]
Any status with this? Can I help with something? @hone

**Comment by pichot** [comment]
Another bump on status. 🙌 This would be fantastic to get in. If not, anyone have a good working fork?

**Comment by steve21168** [comment]
Agreed this is a feature we need.  Can we get this merged in?

**Comment by miguelcobain** [comment]
I wish this could be merged.

**Comment by miguelcobain** [comment]
If someone could let me know the direction this PRs needs to take, I'd be happy to work on it to get it merged.

**Comment by joshwlewis** [comment]
This buildpack has been deprecated. If this is a feature you need, I'd suggest implementing it your own fork and using that instead.
